doc/server/configuration.md

```diff
@@ -0,0 +1,336 @@
+# Limitador configuration
+
+## Command line configuration
+
+The preferred way of starting and configuring the Limitador instance is using the command line:
+
+```
+USAGE:
+    limitador-server [OPTIONS] <LIMITS_FILE> [STORAGE]
+
+ARGS:
+    <LIMITS_FILE>    The limit file to use
+
+OPTIONS:
+    -b, --rls-ip <ip>              The IP to listen on for RLS [default: 0.0.0.0]
+    -p, --rls-port <port>          The port to listen on for RLS [default: 8081]
+    -B, --http-ip <http_ip>        The IP to listen on for HTTP [default: 0.0.0.0]
+    -P, --http-port <http_port>    The port to listen on for HTTP [default: 8080]
+    -l, --limit-name-in-labels     Include the Limit Name in prometheus label
+    -v                             Sets the level of verbosity
+        --validate                 Validates the LIMITS_FILE and exits
+    -h, --help                     Print help information
+    -V, --version                  Print version information
+
+STORAGES:
+    memory          Counters are held in Limitador (ephemeral)
+    redis           Uses Redis to store counters
+    redis_cached    Uses Redis to store counters, with an in-memory cache
+```
+
+The values used are authoritative over any [environment variables](#configuration-using-environment-variables) independently set.
+
+### Limit definitions
+
+The `LIMITS_FILE` provided is the source of truth for all the limits that will be enforced. The file location will be 
+monitored by the server for any changes and be hot reloaded. If the changes are invalid, they will be ignored on hot
+reload, or the server will fail to start. 
+
+#### The `LIMITS_FILE`'s format
+
+When starting the server, you point it to a `LIMITS_FILE`, which is expected to be a _yaml_ file with an array of
+`limit` definitions, with the following format:
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "name": {
+      "type": "string"
+    },
+    "namespace": {
+      "type": "string"
+    },
+    "seconds": {
+      "type": "integer"
+    },
+    "max_value": {
+      "type": "integer"
+    },
+    "conditions": {
+      "type": "array",
+      "items": [
+        {
+          "type": "string"
+        }
+      ]
+    },
+    "variables": {
+      "type": "array",
+      "items": [
+        {
+          "type": "string"
+        }
+      ]
+    }
+  },
+  "required": [
+    "namespace",
+    "seconds",
+    "max_value",
+    "conditions",
+    "variables"
+  ]
+}
+```
```

Comment:
Although it doesn't change much, it could also be stated as a [YAML schema](https://asdf-standard.readthedocs.io/en/1.6.0/schemas/yaml_schema.html):

```yaml
---
"$schema": http://json-schema.org/draft-04/schema#
type: object
properties:
  name:
    type: string
  namespace:
    type: string
  seconds:
    type: integer
  max_value:
    type: integer
  conditions:
    type: array
    items:
    - type: string
  variables:
    type: array
    items:
    - type: string
required:
- namespace
- seconds
- max_value
- conditions
- variables
```